x-pack/filebeat/input/entityanalytics/provider/azuread/azure.go

```diff
@@ -165,6 +165,21 @@ func (p *azure) runFullSync(inputCtx v2.Context, store *kvstore.Store, client be
 		tracker.Wait()
 	}
 
+	if len(state.devices) != 0 {
+		tracker := kvstore.NewTxTracker(ctx)
+
+		start := time.Now()
+		p.publishMarker(start, start, inputCtx.ID, true, client, tracker)
+		for _, d := range state.devices {
+			p.publishDevice(d, state, inputCtx.ID, client, tracker)
+		}
+
+		end := time.Now()
+		p.publishMarker(end, end, inputCtx.ID, false, client, tracker)
+
+		tracker.Wait()
+	}
+
```

Comment:
Should we consider a debug log if the devices length is > 0 ? Just thinking about troubleshooting this part without any log info?